### PR TITLE
DeepMask

### PR DESCRIPTION
**Pull request by Dref360**

### CODE REVIEW

File: deep-learning/_posts/2017-04-10-deepmask.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+layout: review
+title:  "Learning to Segment Object Candidates"
+tags:   deep-learning CNN segmentation
+author: Frédéric Branchaud-Charron
+pdf:   "https://arxiv.org/pdf/1612.03144.pdf"
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-04-10-deepmask.md]: You forgot to change the pdf link

File: deep-learning/_posts/2017-04-10-deepmask.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+<div align="middle">
+  <img src="/deep-learning/images/deepmask/architecture.png" width="600">
+</div>
+
+
+
+The authors propose two models, DeepMask and DeepMaskZoom. The later uses smaller scales to improve performance on smaller objects. This comes at a cost of increased inference time (not provided by the authors). DeepMask runs at 1.6s by image.
+
+## Training
+The authors only trained on positive patches. This means that there is an object in every training patches. This does not mean that the object is centered.
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-04-10-deepmask.md]: every training patch

File: deep-learning/_posts/2017-04-10-deepmask.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+title:  "Learning to Segment Object Candidates"
+tags:   deep-learning CNN segmentation
+author: Frédéric Branchaud-Charron
+pdf:   "https://arxiv.org/pdf/1612.03144.pdf"
+cite:
+  authors: "Pinheiro, P. O., Collobert, R., & Dollar, P."
+  title:   "Learning to segment object candidates"
+  venue:   "Advances in Neural Information Processing Systems (pp. 1990-1998) (2015)"
+---
+
```
Comment (lemairecarl) [inline deep-learning/_posts/2017-04-10-deepmask.md]: Suggestion: give a brief overview of the purpose of the method before anything else. For example:

>This paper presents the DeepMask architecture, which can be used as an object detection method when combined with R-CNN.

File: deep-learning/_posts/2017-04-10-deepmask.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+---
+layout: review
+title:  "Learning to Segment Object Candidates"
```
Comment (jodoin) [inline deep-learning/_posts/2017-04-10-deepmask.md]: DeepMask : Learning to Segment Object Candidates

File: deep-learning/_posts/2017-04-10-deepmask.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+pdf:   "https://arxiv.org/pdf/1506.06204.pdf"
+cite:
+  authors: "Pinheiro, P. O., Collobert, R., & Dollar, P."
+  title:   "Learning to segment object candidates"
+  venue:   "Advances in Neural Information Processing Systems (pp. 1990-1998) (2015)"
+---
+
+This paper presents the DeepMask architecture, which can be used as an object detection method when combined with R-CNN.
+
+DeepMask works on a patch-based approach. For each patch, the model tries to predict two things. First of all, the model will output a pixel-wise segmentation map to find salient objects. Then, in another branch, the model will predict if the patch is at the center of the object. This allows overlapping objects to be segmented correctly.
```
Comment (jodoin) [inline deep-learning/_posts/2017-04-10-deepmask.md]: will output -> outputs
will predict -> predicts

this allows overlapping objects of the same class to be segmented independently.

**Pull request merged**